internal/policy/policy.go

```diff
@@ -13,31 +12,30 @@ const (
 	PolicyStagingRef = "refs/gittuf/policy-staging"
 )
 
+var ErrPolicyExists = errors.New("cannot initialize Policy namespace as it exists already")
+
 // InitializeNamespace creates a git ref for the policy. Initially, the entry
 // has a zero hash.
-// Note: policy.InitializeNamespace assumes the gittuf namespace has been
-// created already.
-func InitializeNamespace() error {
-	repoRootDir, err := common.GetRepositoryRootDirectory()
-	if err != nil {
-		return err
+func InitializeNamespace(repo *git.Repository) error {
+	if _, err := repo.Reference(plumbing.ReferenceName(PolicyRef), true); err != nil {
+		if !errors.Is(err, plumbing.ErrReferenceNotFound) {
+			return err
+		}
+	} else {
+		return ErrPolicyExists
 	}
 
-	refPaths := []string{
-		filepath.Join(repoRootDir, common.GetGitDir(), PolicyRef),
-		filepath.Join(repoRootDir, common.GetGitDir(), PolicyStagingRef),
-	}
-	for _, refPath := range refPaths {
-		if _, err := os.Stat(refPath); err != nil {
-			if os.IsNotExist(err) {
-				if err := os.WriteFile(refPath, plumbing.ZeroHash[:], 0644); err != nil {
-					return err
-				}
-			} else {
-				return err
-			}
+	if _, err := repo.Reference(plumbing.ReferenceName(PolicyStagingRef), true); err != nil {
+		if !errors.Is(err, plumbing.ErrReferenceNotFound) {
+			return err
 		}
+	} else {
+		return ErrPolicyExists
+	}
+
+	if err := repo.Storer.SetReference(plumbing.NewHashReference(plumbing.ReferenceName(PolicyRef), plumbing.ZeroHash)); err != nil {
```

Comment:
You can wrap this in a loop to clean this up a tad:

```go
for _, name := []string{PolicyRef, PolicyStagingRef} {
  if _, err := repo.Reference(plumbing.ReferenceName(PolicyRef), true); err != nil {
		if !errors.Is(err, plumbing.ErrReferenceNotFound) {
			return err
		}
	} else {
		return ErrPolicyExists
	}
}
```

Reply:
Done.